src/lib.rs

```diff
@@ -4442,6 +4443,82 @@ pub fn digraph_spiral_layout(
     layout::spiral_layout(&graph.graph, scale, center, resolution, equidistant)
 }
 
+fn _num_shortest_paths_unweighted<Ty: EdgeType>(
+    graph: &StableGraph<PyObject, PyObject, Ty>,
+    source: usize,
+) -> HashMap<usize, u128> {
+    let mut out_map: Vec<Option<u128>> = vec![None; graph.node_bound()];
+    let node_index = NodeIndex::new(source);
+    let mut bfs = Bfs::new(&graph, node_index);
+    let mut distance: Vec<Option<u128>> = vec![None; graph.node_bound()];
+    distance[node_index.index()] = Some(0);
+    out_map[source] = Some(1);
+    while let Some(current) = bfs.next(graph) {
+        let new_distance = distance[current.index()].unwrap() + 1;
+        for neighbor_index in
+            graph.neighbors_directed(current, petgraph::Direction::Outgoing)
+        {
+            let neighbor: usize = neighbor_index.index();
+            if distance[neighbor].is_none()
+                || distance[neighbor] > Some(new_distance)
+            {
+                distance[neighbor] = Some(new_distance);
+                out_map[neighbor] = out_map[current.index()];
+            } else if distance[neighbor] == Some(new_distance) {
+                out_map[neighbor] = Some(
+                    out_map[neighbor].unwrap()
+                        + out_map[current.index()].unwrap(),
+                );
+            }
+        }
+    }
+    out_map[source] = None;
+    out_map
+        .into_iter()
+        .enumerate()
+        .filter(|i| i.1.is_some())
+        .map(|x| (x.0, x.1.unwrap()))
+        .collect()
```

Comment:
Nodes that are not reachable from `source` will not be included in the final map. So either we state this clearly in the docs or we add them in the `out_map` with a zero value.

Reply:
Sure, I'll update the documentation for the function to explicitly call this out. I think either is a valid approach but at least to me it makes a bit more sense to just not include it in the output map if there is no path from `source` to a node.

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/355/commits/30d5a47f21130ce8708029f5e9b8d2206f0026cd